portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +28,33 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+var slideIndex = 1;
+showSlides(slideIndex);
+
+// Next/previous controls
+function plusSlides(n) {
+  showSlides(slideIndex += n);
+}
+
+// Thumbnail image controls
+function currentSlide(n) {
+  showSlides(slideIndex = n);
+}
+
+function showSlides(n) {
+  var i;
```

Comment:
Not a big deal since I'm late. But I'd add put some line breaks in here (or on the other projects) to improve readability, for instance separating the assignment operations from the more complex if/for blocks.